lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
I think what I don't like about  `QueryRelation.new(self) { |mode, options| search(mode, options) }` is that the block is a pointless forwarding method.  Declaring that search is the default method, and providing an override (see the query_method comment above) felt more Rubyish.

I see your point about it being semi-magical that QueryRelation is going to, by default, call `search` on `self`, and it appears you want to make that more explicit.  I was sort of taking the documentation-driven approach that you see with things like Mixins.  For example, it is documented for `Enumerable` that if you implement the `.each` method, then everything will just work.  Similarly, I was thinking that for `QueryRelation` if you implement the `.search` method, then everything will just work.

Note that if the caller wants explicit, they can also do `QueryRelation.new(self, :query_method => :search)`.  Does that work for you?